### ISSUE TITLE
Idea: Mixed fraction literal

### ISSUE THREAD

**Issue by shirok** [shirok/Gauche#804]
It is handy when you're using REPL as a calculator---esp. in US, measurements are often expressed in mixed fractions.

E.g. `1+3/4` may be read as a rational number, equivalent to `7/4`.

Issue: `1+3/4` doesn't go well with complex numbers.

**Comment by lassik** [comment]
Or just `(infix 1+3/4)`? We could make an infix math SRFI some time (I won't have time this year, but would be interesting).

**Comment by shirok** [comment]
Infix syntax is interesting, but this is more for a throwaway calculation on REPL, so a terse syntax is better. I was dealing with a woodworking book then and there are computations such as 19+1/2 inches plus 1+3/4  inches everywhere :-)

**Comment by lassik** [comment]
See also [potential numbers](http://www.lispworks.com/documentation/HyperSpec/Body/02_caa.htm) in Common Lisp.

**Comment by jpellegrini** [comment]
> Issue: `1+3/4` doesn't go well with complex numbers.

The reader can read the two numbers, and only decide if they're a complex or a mixed fraction after reading enough...

`1/2-3/4i` is complex  (read `1/2`, then `-3/4`, then `i` -- the two rationals were kept in two pre-defined buffers before deciding if what's being read is complex or not)

`1/2-3/4` should be split into 2 tokens, `1/2` and `-3/4` (and that's probably an error by whoever typed it :)  I suppose the easy way would be to, when noticing this situation, just return the `1/2` token and rewind the cursor (however it's implemented) to start reading from the `-` sign.

`2-3/4i` is complex (same as the first case)

`2-3/4` is a mixed fracion (read `2`, then `-3/4`, and _since there's no `i` in the end, and the first is integer and the second is rational_, then it's a mixed fraction)

The decision of what type the number is would just be delayed a tiny bit... No?

**Comment by shirok** [comment]
@jpellegrini Yes technically it can be distinguished from complex numbers, but having syntaxes that looks similar but meaning different things is a source of confusion and/or error.

Parsing `1/2-3/4' as two tokens is a diversion of current convention.  If we allow it, I'm afraid people start assuming things look like an infix traditional mathematical expression is parsed as an infix expression.  I don't like that.

Having `1+2/3` and `1+2/3i` is more subtle, for both can be taken as a signle token of a numeric constant.  My concern is twofold: People who are familiar with Scheme syntax may think 1+2/3 is an error of 1+2/3i and be confused; people who aer not familiar with Scheme syntax may thing if 1+2/3 is allowed then why not 1/2+2/3.

Generally the decision needs to compare the advantage of having the new syntax and the possible complexity/confusion caused by it.  I'm still not sure the balance regarding of this issue.   If I see having 1+2/3 is clearly an advantage, then I'll add it.
